Content.Server/DeltaV/Cargo/Systems/StockMarketSystem.cs

```diff
@@ -110,109 +102,85 @@ private void OnStockTradingMessage(Entity<StockTradingCartridgeComponent> ent, r
         }
         finally
         {
             // Raise the event to update the UI regardless of outcome
-            var ev = new StockMarketUpdatedEvent(station.Value);
-            RaiseLocalEvent(ev);
+            UpdateStockMarket(station);
         }
     }
 
-    private bool TryBuyStocks(
+    private void UpdateStockMarket(EntityUid station)
+    {
+        var ev = new StockMarketUpdatedEvent(station);
+        RaiseLocalEvent(ref ev);
+    }
+
+    private bool TryChangeStocks(
         EntityUid station,
         StationStockMarketComponent stockMarket,
         int companyIndex,
-        int amount)
+        int amount,
+        EntityUid user)
     {
-        if (amount <= 0 || companyIndex < 0 || companyIndex >= stockMarket.Companies.Count)
+        if (amount == 0 || companyIndex < 0 || companyIndex >= stockMarket.Companies.Count)
             return false;
 
         // Check if the station has a bank account
         if (!TryComp<StationBankAccountComponent>(station, out var bank))
             return false;
 
         var company = stockMarket.Companies[companyIndex];
-        var totalValue = (int)Math.Round(company.CurrentPrice * amount);
-
-        // See if we can afford it
-        if (bank.Balance < totalValue)
-            return false;
+        var totalValue = (int)Math.Round(company.CurrentPrice * (float) amount);
 
         if (!stockMarket.StockOwnership.TryGetValue(companyIndex, out var currentOwned))
             currentOwned = 0;
 
-        // Update the bank account
-        _cargo.UpdateBankAccount(station, bank, -totalValue);
-        stockMarket.StockOwnership[companyIndex] = currentOwned + amount;
-
-        // Log the transaction
-        _adminLogger.Add(LogType.Action,
-            LogImpact.Medium,
-            $"[StockMarket] Bought {amount} stocks of {company.LocalizedDisplayName} at {company.CurrentPrice:F2} credits each (Total: {totalValue})");
-
-        return true;
-    }
-
-    private bool TrySellStocks(
-        EntityUid station,
-        StationStockMarketComponent stockMarket,
-        int companyIndex,
-        int amount)
-    {
-        if (amount <= 0 || companyIndex < 0 || companyIndex >= stockMarket.Companies.Count)
-            return false;
-
-        // Check if the station has a bank account
-        if (!TryComp<StationBankAccountComponent>(station, out var bank))
-            return false;
-
-        if (!stockMarket.StockOwnership.TryGetValue(companyIndex, out var currentOwned) || currentOwned < amount)
-            return false;
-
-        var company = stockMarket.Companies[companyIndex];
-        var totalValue = (int)Math.Round(company.CurrentPrice * amount);
+        if (amount > 0)
+        {
+            // Buying: see if we can afford it
+            if (bank.Balance < totalValue)
+                return false;
+        }
+        else
+        {
+            // Selling: see if we have enough stocks to sell
+            var selling = -amount;
```

Comment:
remove

Reply:
its meant to be used just wrong var name :trollface: